### ISSUE TITLE
Improve the search function to include "é", "è" as a search for "e" and vice versa

### ISSUE THREAD

**Issue by Massedil** [thundernest/k-9#6503]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.400

### Where did you get the app from?

F-Droid

### Android version

LineageOS 18.1 / Android 11

### Device model

Samsung Galaxy S5 / Samsung Galaxy S7

### Steps to reproduce

In one empty folder, put an email with a word containing a "é" in the `Subjet`, like "école".

Now search this folder with "ecole".

### Expected behavior

The email with "école" in the subject is found.

The same change permit a search with "a" for "à", etc.

### Actual behavior

No email is found.

**Comment by kewisch** [comment]
There is an unicode algorithm for this, no need to reinvent the wheel :)

**Comment by kv1dr** [comment]
@kewisch, you can assign this to me. I already start looking at it :)

**Comment by kv1dr** [comment]
According to stackoverflow, you can't filter columns by "unaccent" characters with SQL where clause, at least not in any clean way. The cleanest way, that was recommended by [this stackoverflow answer](https://stackoverflow.com/a/16283863/561785), was to add another column to table, that would contain normalized value of original column. E.g. **subject** would contain **école**, while **subject_normalized** would contain **ecole**. And we would filter by **subject_normalized**, but display  **subject** and the same would go for content.

So few things would need to be done:

- every time new mail is saved to database, we would need to normalize subject and content (would we need to also normalize other columns?)
- make migration to normalize already saved mails
- I probably forget something...

Any comments on that? Would this be acceptable solution?

**Comment by kewisch** [comment]
It seems to me the two options are either manually normalizing the fields we are searching in, or using a sqlite FTS extension that allows tokenizing in a way that ignores diacritics. The migration could potentially be very costly if you have a lot of email.

When I made this a good first bug I was kind of hoping we could simply change this in the sqlite query, if it requires a more elaborate solution then it might make sense to rather do this as part of #8834 where we're intending to revamp the database completely to allow for better synchronization options.

@wmontwe might have technical opinions on this as well.

**Comment by kv1dr** [comment]
I agree, migration could take a lot of time, so it doesn't really look like a viable option. I didn't know about sqlite FTS extensions. By quick google search it looks like it is doable. I will research a little bit more, try to implement it and report back my findings. Unfortunately, I am going on vacation now, so I won't be able to start working on this until the middle of April.

**Comment by kewisch** [comment]
I believe the FTS might also require some migration, since the content needs to be tokenized. When you are back feel free to do some research on that and we can find out together if we want to tack it on to the existing database or not. Enjoy your time off!

**Comment by kewisch** [comment]
@kv1dr reached out separately and won't have time right this moment due to other obligations. If someone wants to pick this up I'd highly recommend previous experience with FTS migrations.
